src/onegov/api/models.py

```diff
@@ -217,16 +234,30 @@ def item_links(self, item):
         raise NotImplementedError()
 
     @property
-    def collection(self):
+    def collection(self) -> 'PaginatedGenericCollection[_M]':
```

Comment:
This implies to me that `ApiEndpoint` should be a `Generic` in `_M` as well, that way you can also improve `by_item` `by_id` and `batch` and can replace `Any` with `_M`. In this case `_M` is unbound anyways, so it's only slightly better than `PaginatedGenericCollection[Any]`

Reply:
Something like `Intersection[FooClass, BarMixin]` would be pleasant to have, but I think I'll remove this `PaginatedGenericCollection` and use a Protocol.